### DISCUSSION TITLE
Is there a license issue with the minio Gateway used

### DISCUSSION THREAD

**Discussion by yelog** [juicedata/juicefs#1351] [Q&A] [answered]
Because Minio uses the AGPL-3.0 License,  this may cause legal woes for many.

My Object Storage is not Minio, but I run a gateway through the command `juicefs gateway ...`. Is there the license issue above.

If anyone knows, please answer it. Thx!

**Comment by davies** [comment, marked as answer]
The version [1] of MinIO is released under Apache License v2, so there is not GPL issue here. 

The problem is that we can't upgrade to latest version of MinIO. We plan to add a plugin of JuiceFS into MinIO.

[1] https://github.com/juicedata/minio
